pkg/runtime/function/cast_date.go

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package function
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+import (
+	"github.com/pkg/errors"
+)
+
+import (
+	"github.com/arana-db/arana/pkg/proto"
+)
+
+// FuncCastDate is  https://dev.mysql.com/doc/refman/5.6/en/cast-functions.html#function_cast
+const FuncCastDate = "CAST_DATE"
+
+var DateSep = "~!@#$%^&*_+=:;,.|/?\\(\\)\\[\\]\\{\\}\\-\\\\"
+var _ proto.Func = (*castDateFunc)(nil)
+
+func init() {
+	proto.RegisterFunc(FuncCastDate, castDateFunc{})
+}
+
+type castDateFunc struct{}
+
+func (a castDateFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
+	// expr
+	val, err := inputs[0].Value(ctx)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	dateArgs := fmt.Sprint(val)
+	if strings.Compare(dateArgs, "") == 0 {
+		return a.DefaultDateValue(), nil
+	}
+
+	// format - YY-MM-DD, YYYY-MM-DD
+	pat := "^\\d{1,4}[" + DateSep + "]+\\d{1,2}[" + DateSep + "]+\\d{1,2}$"
+	match, err := regexp.MatchString(pat, dateArgs)
```

Comment:
It will be better if pre-compile the regexp as a global variable.

Reply:
done